File: kotlinTutorials/kt10/src/main/java/com/java/kt10/LazyThreadSafeStaticInnerClass.java
package com.java.kt10;

/**
 * 静态内部类
 */
public class LazyThreadSafeStaticInnerClass {

    private static class Holder {
        private static LazyThreadSafeStaticInnerClass instance =
                new LazyThreadSafeStaticInnerClass();
    }

    private LazyThreadSafeStaticInnerClass() {}

    public static LazyThreadSafeStaticInnerClass getInstance() {
       return Holder.instance;
    }
}

File: kotlinTutorials/kt10/src/main/kotlin/com/kotlin/kt10/LazyThreadSafeStaticInnerClass.kt
package com.kotlin.kt10

class LazyThreadSafeStaticInnerClass {
    companion object {
        fun getInstance() : LazyThreadSafeStaticInnerClass {
            return Holder.instance
        }
    }

    private object Holder {
        var instance = LazyThreadSafeStaticInnerClass()
    }
}